File: app/src/main/java/me/topas/rssreader/MainContract.java
package me.topas.rssreader;

import me.topas.rssreader.base.BaseView;

/**
 * Created by ftoptas on 28/01/17.
 */

public interface MainContract {

    // User actions. Presenter will implement
    interface Presenter {
        void loadHelloText();
    }

    // Action callbacks. Activity/Fragment will implement
    interface View extends BaseView {
        void onTextLoaded(String text);
    }

}

File: app/src/main/java/me/toptas/rssreader/features/main/MainContract.kt
package me.toptas.rssreader.features.main

import me.toptas.rssreader.base.BaseMvpPresenter
import me.toptas.rssreader.base.BaseView
import me.toptas.rssreader.model.Feed

/**
 * Created by ftoptas on 28/01/17.
 */

interface MainContract {

    // User actions. Presenter will implement
    interface Presenter : BaseMvpPresenter<View> {
        fun loadRssFragments()

    }

    // Action callbacks. Activity/Fragment will implement
    interface View : BaseView {
        fun onLoadRssFragments(feeds: List<Feed>)
    }

}
